### PR TITLE
Refactor and mocking unit test for bigquery.py

### PR DESCRIPTION
**Pull request by hirosassa**
## What is this PR

I refactored and applied mocks on unit test for bigquery.py.
What I did are:
1. removed deprecated methods (e.g. `google.cloud.bigquery.Client.dataset` )
2. moved some procedures into private functions for testability
3. applied mock on UT and divide GCP integration test and Local unit test

### CODE REVIEW

File: src/lib/bigquery.py [hunk 1]
```diff
@@ -63,85 +58,81 @@ def update_dataset_desc(self, dataset_desc: DatasetDesc) -> BqUpdateResult:
             existing_dataset_desc: DatasetDesc = self.get_dataset_desc(dataset_id=dataset_id)
             if existing_dataset_desc.description == dataset_desc.description:
                 return BqUpdateResult(True, ResultType.SAME, detail="do nothing")
-            else:
```
Comment (hirosassa) [inline src/lib/bigquery.py:66]: This `else` block is unnecessary because `if` block ends with `return`

File: src/lib/bigquery.py [hunk 2]
```diff
@@ -41,19 +41,14 @@ def __init__(self, config, logger):
...
-        return ret
+        return [project.project_id for project in self.client.list_projects() if _filter_by_patterns(project.project_id, include_pattern, exclude_pattern)]
 
     # -------------------------------
     # Dataset
     # -------------------------------
 
     def get_dataset_desc(self, dataset_id) -> DatasetDesc:
-        dataset_ref = self.client.dataset(dataset_id)
-        dataset = self.client.get_dataset(dataset_ref)  # API Request Here
```
Comment (hirosassa) [inline src/lib/bigquery.py:56]: removed deprecated method

File: src/lib/bigquery.py [hunk 3]
```diff
@@ -63,85 +58,81 @@ def update_dataset_desc(self, dataset_desc: DatasetDesc) -> BqUpdateResult:
...
+            return BqUpdateResult(True, ResultType.UPDATE, f"{existing_dataset_desc.description} -> {dataset_desc.description}")
+
         except NotFound:
             if self.config.ignore_dataset_not_found_error_when_restore:
-                is_success = True
-            else:
-                is_success = False
-            return BqUpdateResult(is_success, ResultType.DATASET_NOT_FOUND)
+                return BqUpdateResult(True, ResultType.DATASET_NOT_FOUND)
+            return BqUpdateResult(False, ResultType.DATASET_NOT_FOUND)
```
Comment (hirosassa) [inline src/lib/bigquery.py:68]: simplified nested logic

File: test/test_bigquery.py [hunk 4]
```diff
@@ -15,17 +17,35 @@
...
+def dummy_projects():
+    dummy_project1 = mock.MagicMock(project_id='dummy_project1')
+    dummy_project2 = mock.MagicMock(project_id='dummy_project2')
+    return [dummy_project1, dummy_project2]
+
+
 @pytest.mark.gcp_project
-class TestBigquery(unittest.TestCase):
-    def setUp(self):
+class TestBigqueryGCP(unittest.TestCase):
```
Comment (hirosassa) [inline test/test_bigquery.py:40]: divide into `TestBigqueryGCP` using actual GCP project, and `TestBigqueryLocal` for local testing without GCP project

**Review by TetsutaroWatanabe — APPROVED** [review]
LGTM

**Pull request merged**